src/Features/Core/Portable/Shared/Naming/IdentifierNameParts.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.NamingStyles;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Shared.Naming
+{
+    internal readonly struct IdentifierNameParts
+    {
+        public readonly string BaseName;
+        public readonly ImmutableArray<string> BaseNameParts;
+
+        public IdentifierNameParts(string baseName, ImmutableArray<string> baseNameParts)
+        {
+            BaseName = baseName;
+            BaseNameParts = baseNameParts;
+        }
+
+        public static IdentifierNameParts GetIdentifierBaseName(ISymbol symbol, ImmutableArray<NamingRule> rules)
+        {
+            var baseName = RemovePrefixesAndSuffixes(symbol, rules, symbol.Name);
+
+            var parts = StringBreaker.GetWordParts(baseName);
+            var result = CreateWords(parts, baseName);
```

Comment:
I was confused by the name `result` here... can you rename to `words` (or similar) or just inline the call?

Reply:
fixed